tests/test_extension.py

```diff
@@ -0,0 +1,103 @@
+# -*- coding: utf-8 -*-
+
+import os
+import random
+import string
+from contextlib import contextmanager
+
+import pytest
+from kafka import SimpleClient, SimpleConsumer
+from scrapy.utils.serialize import ScrapyJSONDecoder
+from scrapy.utils.test import get_crawler
+from scrapy_kafka_export import KafkaItemExporterExtension
+from scrapy_kafka_export.writer import UnknownTopicError
+
+
+def random_string(l):
+    return "".join(random.choice(string.ascii_letters) for i in range(l))
+
+def consume_from_kafka(client, topic):
+    consumer = SimpleConsumer(client=client, group=None, topic=topic)
+    consumer.seek(-1, 2)
+    msg = consumer.get_message(block=False, timeout=10)
+    consumer.stop()
+    if msg:
+        deserializer = ScrapyJSONDecoder()
+        return deserializer.decode(msg.message.value.decode('utf8'))
+    raise Exception
+
+def run_command(command):
+    import subprocess
+    return subprocess.check_output(['bash','-c', command])
+
+def create_topic(topic):
+    bash_command = "{}/kafka-topics.sh --create " \
+                   "--zookeeper {} --replication-factor 1 " \
+                   "--partitions 1 --topic {}".format(
+        os.getenv('KAFKA_BIN'), os.getenv('ZOOKEEPER'), topic
+    )
+    return run_command(bash_command)
+
+def delete_topic(topic):
+    bash_command = "{}/kafka-topics.sh --delete " \
+                   "--zookeeper {} --topic {}".format(
+        os.getenv('KAFKA_BIN'), os.getenv('ZOOKEEPER'), topic
+    )
+    return run_command(bash_command)
+
+@contextmanager
+def opened_middleware(crawler):
+    mw = KafkaItemExporterExtension.from_crawler(crawler)
+    crawler.spider = crawler._create_spider('example')
+    mw.spider_opened(crawler.spider)
+    try:
+        yield mw
+    finally:
+        mw.spider_closed(crawler.spider)
+
+class TestClass:

```

Comment:
Usually test cases are named according to functionality they test; e.g. here TestKafkaExtension or TestKafkaImporterExtension can be a better name.